src/components/ReportActionItem/MoneyRequestAction.js

```diff
@@ -66,11 +78,40 @@ const defaultProps = {
     iouReport: {},
     reportActions: {},
     isHovered: false,
+    session: {
+        email: null,
+    },
 };
 
 const MoneyRequestAction = (props) => {
     const hasMultipleParticipants = props.chatReport.participants.length > 1;
     const onIOUPreviewPressed = () => {
+        // This would ideally be passed as a prop or hooked up via withOnyx so that we are not be triggering a potentially intensive
+        // search in an onPress handler, I think this could lead to performance issues but it probably ok for now.
+        const thread = getThreadForReportActionID(props.action.reportActionID);
+        console.log({thread});
+        if (_.isEmpty(thread)) {
+            // Since a thread does not exist yet then we need to create it now. This is done by creating the report object
+            // and passing the parentReportActionID of the reportAction. OpenReport will then automatically create the thread for us.
+            const optimisticThreadReport = buildOptimisticChatReport(
+                props.chatReport.participants,
+                props.translate('iou.threadReportName', {payee: props.action.actorEmail, comment: props.action.originalMessage.comment}),
+                '',
+                props.chatReport.policyID,
+                props.chatReport.owner,
+                props.chatReport.type === CONST.REPORT.CHAT_TYPE.POLICY_EXPENSE_CHAT,
+                props.chatReport.oldPolicyName,
+                undefined,
+                CONST.REPORT.NOTIFICATION_PREFERENCE.ALWAYS,
+                props.action.reportActionID,
+            );
+            console.log({optimisticThreadReport});
+            Report.openReport(
+                optimisticThreadReport.reportID,
+                [_.reject(optimisticThreadReport.participants, (login) => login === lodashGet(props.session, 'email', ''))],
```

Comment:
Should this be the same as `props.chatReport.participants`?

Reply:
I wasn't 100% sure since the API param for participants says it shouldn't include the user making the request: https://github.com/Expensify/Web-Expensify/blob/335001f0f64bf449fc884456ef1911cd1e6f17d5/lib/ReportAPI.php#L4404-L4405

Reply:
okay asked about this here https://expensify.slack.com/archives/C02HWMSMZEC/p1683912289696679?thread_ts=1683911072.385549&cid=C02HWMSMZEC

And looks like it doesn't really matter if we include the user making the request here because in Auth here we make sure to include them and since `set` doesn't allow duplicates we are fine https://github.com/Expensify/Auth/blob/dbd779f7c423a82d4871d6dc076ecb37406fb43c/auth/command/OpenReport.cpp#L77